tests/test_api.py

```diff
@@ -1913,29 +1913,32 @@ def test_do_create_prod_build_branch_required(self, osbs, branch_name):
                                            customize_conf=customize_conf)
 
     # osbs is a fixture here
-    @pytest.mark.parametrize(('name', 'data'), [  # noqa
-        ('special-config',
-         {
-             "special.how": "very",
-             "special.type": "charm"
-         }),
-    ])
-    def test_config_map(self, osbs, name, data):
+    def test_config_map(self, osbs): # noqa
         obj = {
             'apiVersion': 'v1',
             'kind': 'ConfigMap',
-            'metadata': {'name': name},
-            'data': data,
+            'metadata': {}
         }
-        (flexmock(osbs.os)
-            .should_call('create_config_map')
-            .with_args(obj))
+        mock = flexmock(sys.modules['__builtin__' if six.PY2 else 'builtins'])
+        mock.should_call('open')  # set the fall-through
+        (mock.should_receive('open')
+            .with_args('inputs/config_map.json')
+            .and_return(flexmock(read=lambda: json.dumps(obj))))
```

Comment:
so I have that file as fp from here: 
   with open(os.path.join(INPUTS_PATH, "config_map.json")) as fp:

I can't just pass it to .and_return like this:
   .and_return(flexmock(read=lambda: fp.read() ))

because it's not a json object that that point. 

So I'm still confused here.